site/stripe/DonateForm.tsx

```diff
@@ -197,161 +202,137 @@ export class DonateForm extends React.Component {
-                                        !this.isCustom
-                                    }
-                                />
-                                <label htmlFor={`amount-${amount}`}>
-                                    {this.currencySymbol}
-                                    {amount}
-                                </label>
-                            </div>
+                                        !this.customAmount,
+                                })}
+                                key={`${amount}-${this.interval}`}
+                            />
                         ))}
-                        <div className="owid-radio custom-radio">
+                        <div
+                            className={cx("donation-custom-amount", {
+                                active: !!this.customAmount,
+                            })}
+                        >
+                            <label htmlFor="donation-custom-amount__input">
+                                {this.currencySymbol}
+                            </label>
                             <input
-                                type="radio"
-                                id="custom"
-                                name="amount"
-                                checked={this.isCustom}
+                                type="text"
+                                placeholder="Other"
+                                id="donation-custom-amount__input"
+                                className="donation-custom-amount__input"
                                 onChange={(event) =>
-                                    this.setIsCustom(event.target.checked)
+                                    this.setCustomAmount(event.target.value)
                                 }
+                                value={this.customAmount}
                             />
-                            <label htmlFor="custom">
-                                {this.currencySymbol}
-                                <input
-                                    type="text"
-                                    placeholder="Other"
-                                    name="custom-amount"
-                                    className="custom-amount-input"
-                                    onChange={(event) =>
-                                        this.setCustomAmount(event.target.value)
-                                    }
-                                    value={this.customAmount}
-                                />
-                            </label>
                         </div>
                     </div>
                 </fieldset>
 
-                <fieldset>
-                    <div className="owid-block-field">
-                        <label htmlFor="name">
-                            <h3>Your name (optional)</h3>
+                <fieldset className="donation-public-checkbox">
+                    <Checkbox
```

Comment:
This is very nitpicky, but the rest of the form is _so nicely done_ that I feel compelled to mention that this is the only form element that lacks a focus indicator when navigating by keyboard.